tests/phpunit/integration/Modules/Tag_ManagerTest.php

```diff
@@ -360,6 +361,76 @@ public function test_get_datapoints() {
 		);
 	}
 
+	public function test_get_assets() {
+		$context    = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$tagmanager = new Tag_Manager( $context );
+
+		$assets = $tagmanager->get_assets();
+
+		$this->assertCount( 1, $assets );
+
+		$script = $assets[0];
+		$script->register( $context );
+
+		$dependency = wp_scripts()->registered['googlesitekit-modules-tagmanager'];
+
+		$this->assertEquals( $context->url( 'dist/assets/' ) . 'js/googlesitekit-modules-tagmanager.js', $dependency->src );
+		$this->assertEqualSets(
+			array(
+				'googlesitekit-api',
+				'googlesitekit-data',
+				'googlesitekit-datastore-site',
+				'googlesitekit-modules',
+				'googlesitekit-vendor',
+				'googlesitekit-modules-analytics',
+			),
+			$dependency->deps
+		);
+
+		wp_deregister_script( 'googlesitekit-modules-tagmanager' );
+	}
+
+	public function test_get_assets__no_analytics() {
+		$context    = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$tagmanager = new Tag_Manager( $context );
+
+		// Override the googlesitekit_module_exists filter to ensure the Analytics module is not available.
+		remove_all_filters( 'googlesitekit_module_exists' );
+		add_filter(
+			'googlesitekit_module_exists',
+			function( $exists, $slug ) {
+				return 'analytics' === $slug ? false : true;
+			},
+			10,
+			2
+		);
+
+		$assets = $tagmanager->get_assets();
+
+		$this->assertCount( 1, $assets );
+
+		$script = $assets[0];
+		$script->register( $context );
+
+		$dependency = wp_scripts()->registered['googlesitekit-modules-tagmanager'];
+
+		$this->assertEquals( $context->url( 'dist/assets/' ) . 'js/googlesitekit-modules-tagmanager.js', $dependency->src );
+
+		// Verify the 'googlesitekit-module-analytics' asset is not a dependency.
```

Comment:
How about we use an assertion for this instead?

Reply:
Hmm, if I am following you correctly do you mean something like this?

```php
		$this->assertNotContains(
			'googlesitekit-module-analytics',
			$dependency->deps
		);
```

I would prefer to keep the current assertion as it's more robust, guarding against `$dependency->deps` containing an unexpected value (or being empty etc) via this execution path.

However I have gone ahead and added the `assertNotContains` as well, so as to state the intent in code rather than a comment.